src/System.Private.ServiceModel/tests/Common/Infrastructure/xunit/ConditionAttribute.cs

```diff
@@ -0,0 +1,121 @@
+    public sealed class ConditionAttribute : WcfSkippableAttribute
+    {
+        public string[] Conditions { get; private set; }
+
+        public ConditionAttribute(params string[] conditions)
+        {
+            Conditions = conditions;
+        }
+
+        public override string GetSkipReason(ITestMethod testMethod)
+        {
+            // A null or empty list of conditions is treated as "no conditions",
+            // and the test cases will not be skipped.
+            // Example: [Condition()] or [Condition((string[]) null)]
+            int conditionCount = Conditions == null ? 0 : Conditions.Length;
+            if (conditionCount == 0)
+            {
+                return null;
+            }
+
+            MethodInfo testMethodInfo = testMethod.Method.ToRuntimeMethod();
+            Type testMethodDeclaringType = testMethodInfo.DeclaringType;
+
+            List<string> falseConditions = new List<string>(conditionCount);
+
+            foreach (string entry in Conditions)
+            {
+                string conditionMemberName = entry;
+
+                // Null condition member names are silently tolerated
+                if (string.IsNullOrWhiteSpace(conditionMemberName))
+                {
+                    continue;
+                }
+
+                string[] symbols = conditionMemberName.Split('.');
+                Type declaringType = testMethodDeclaringType;
+
+                if (symbols.Length == 2)
+                {
+                    conditionMemberName = symbols[1];
+                    ITypeInfo type = testMethod.TestClass.Class.Assembly.GetTypes(false).Where(t => t.Name.Contains(symbols[0])).FirstOrDefault();
+                    if (type != null)
+                    {
+                        declaringType = type.ToRuntimeType();
+                    }
+                }
+
+                MethodInfo conditionMethodInfo;
+                if ((conditionMethodInfo = LookupConditionalMethod(declaringType, conditionMemberName)) == null)
+                {
+                    falseConditions.Add(String.Format("Condition \"{0}\" not found in type \"{1}\".",
+                                                      conditionMemberName, testMethodDeclaringType.Name));
+                    continue;
+                }
+
+                // In the case of multiple conditions, collect the results of all
+                // of them to produce a summary skip reason.
+                try
+                {
+                    if (!(bool)conditionMethodInfo.Invoke(null, null))
+                    {
+                        falseConditions.Add(conditionMemberName);
+                    }
+                }
+                catch (Exception exc)
+                {
+                    falseConditions.Add(String.Format("Condition \"{0}\" threw exception \"{1}\".",
+                                                      conditionMemberName, exc.ToString()));
```

Comment:
>exc.ToString() [](start = 75, length = 14)

This is going to output the entire stack trace as part of the string (which will include new lines). The string text suggests you might have intended to just output the exception type and/or it's message. Just wanted to check the behavior matched the intention.